### PR TITLE
Feature/execution

### PR DESCRIPTION
**Pull request by MizukiSonoko**
Add some logic
- Simple currency 
- Simple Messenger

### CODE REVIEW

File: core/service/executor.cpp [hunk 1]
```diff
@@ -53,27 +66,81 @@ namespace executor{
...
             const auto assetName = tx.asset().name();
-            auto senderAsset    = repository::asset::find(  sender,assetName);
-            auto receiverAsset  = repository::asset::find(receiver,assetName);
+
             // **********************************************************************************
-            // * This is Transfer<Asset>'s logic.
+            // * This is Transfer<Asset>'s logic. Tax send logic
             // **********************************************************************************
+            if(tx.asset().value().find("author") != tx.asset().value().end()) {
+                const auto author = tx.asset().value().at("author").valuestring();
```
Comment (l4l) [inline core/service/executor.cpp]: pair `value().find(..)` & `value().at(..)` looks like performing two jobs instead one
Could it be done in one search? like:
```
auto author = tx.asset().value().find("author");
if (author != tx.asset().value().end()) {
  //...
}
```

File: core/service/executor.cpp [hunk 2]
```diff
@@ -53,27 +66,81 @@ namespace executor{
...
+
+                const auto value = tx.asset().value().at("value").valueint();
 
+                auto senderAsset = repository::asset::find(sender, assetName);
+                auto authorAsset = repository::asset::find(author, assetName);
 
+                auto receiverAsset = repository::asset::find(receiver, assetName);
+                if(senderAsset.value().find("value") != senderAsset.value().end() &&
+                   receiverAsset.value().find("value") != receiverAsset.value().end()){
```
Comment (l4l) [inline core/service/executor.cpp]: same here

File: core/service/executor.cpp [hunk 3]
```diff
@@ -53,27 +66,81 @@ namespace executor{
...
+                    auto senderValue = senderAsset.value().at("value").valueint();
+                    auto receiverValue = receiverAsset.value().at("value").valueint();
+                    if(senderValue > value){
+                        (*senderAsset.mutable_value())["value"].set_valueint(senderValue - value);
+                        (*receiverAsset.mutable_value())["value"].set_valueint(receiverValue + value);
+                        (*authorAsset.mutable_value())["value"].set_valueint(value * 0.8);
```
Comment (l4l) [inline core/service/executor.cpp]: What does `0.8` mean? Use named const plz

File: core/service/executor.cpp [hunk 4]
```diff
@@ -53,27 +66,81 @@ namespace executor{
...
+                }
+                repository::asset::update(sender, assetName, senderAsset);
+                repository::asset::update(receiver, assetName, receiverAsset);
+
+            // **********************************************************************************
+            // * This is Transfer<Asset>'s logic. multi message chat
             // **********************************************************************************
-            repository::asset::update(sender,assetName,senderAsset);
-            repository::asset::update(receiver,assetName,receiverAsset);
+            }else if(tx.asset().value().find("targetName") != tx.asset().value().end()){
```
Comment (l4l) [inline core/service/executor.cpp]: this else-if branch are same
Extract that to a separate function and pass different strings so there won't code duplicates

File: core/transaction_builder/helper/create_objects_helper.cpp [hunk 5]
```diff
@@ -119,14 +119,15 @@ Api::Account createAccount(std::string publicKey, std::string name,
...
+Api::Asset createAsset(
+  std::string domain,
+  std::string name,
+  ::txbuilder::Map value, std::string smartContractName) {
   Api::Asset ret;
   ret.set_domain(std::move(domain));
   ret.set_name(std::move(name));
   *ret.mutable_value() = ::google::protobuf::Map<std::string, Api::BaseObject>(
       value.begin(), value.end());
-  ret.set_smartcontractname(std::move(smartContractName));
```
Comment (l4l) [inline core/transaction_builder/helper/create_objects_helper.cpp]: why?
If removing remove redundant param at fn signature as well

Comment (MizukiSonoko) [inline core/transaction_builder/helper/create_objects_helper.cpp]: I want to confirm this branch and develop as v0.75.
And thia version doesn't need smart contract.
so I remove it temporarily

Comment (MizukiSonoko) [inline core/transaction_builder/helper/create_objects_helper.cpp]: I want to confirm this branch and develop as v0.75.
And thia version doesn't need smart contract.
so I remove it temporarily

**Comment by l4l** [comment]
I'm going to resolve conflict

**Pull request merged**